src/Engine/ProtoCore/FFI/CLRDLLModule.cs

```diff
@@ -396,15 +407,27 @@ private ClassDeclNode ParseSystemType(Type type, string alias)
                 if (SupressesImport(m, mGetterAttributes))
                     continue;
 
-                if (classnode.IsStatic && m.GetBaseDefinition().DeclaringType == baseType && baseType == typeof(object))
+                MethodInfo baseDefinition;
+                if (m.DeclaringType.Assembly.ReflectionOnly)
+                {
+                    baseDefinition = m.GetBaseDefinitionReflectionContext();
+                    if (baseDefinition is null)
```

Comment:
@mjkkirschner in `GetBaseDefinitionReflectionContext` we try to get the same method but from the BaseType so essentially checking if it is an overloaded method, issues can happen if there are overloads of that method in the baseType. Normally we could just do `baseType.GetMethod(string methodName, Type[] argumentTypes);` but in reflection, this can become an issue as we can guarantee that the argument types are loaded and therefore we can GetParameters from the type